NewsApp-main/NewsApp/Utilities/Network/Service/NetworkService.swift

```diff
@@ -0,0 +1,53 @@
+//
+//  NetworkService.swift
+//  NewsApp
+//
+//  Created by İsmail Palalı on 26.08.2023.
+//
+
+import Alamofire
+import UIKit
+
+// MARK: - ServiceProtocol
+protocol NetworkServiceProtocol: AnyObject {
+    func fetchNews(completion: @escaping (Result<NewsModel?, APIError>) -> Void)
+    func fetchDetailNews(id: String, completion: @escaping (Result<NewsDetail?, APIError>) -> Void)
+}
+
+// MARK: - Services
+final class NetworkService: NetworkServiceProtocol {
```

Comment:
Bu NetworkService neyin servisi mesela? Belli bir endpoint grubu için mi özelleştireceksin? Yoksa tüm endpointler için mi? Tümü için büyük bir projede bunu yapamazsın her gelen dokunur ve patates olur buralar. O yüzden mantıklı bölmek lazım. 

Mesela SearchAPIService desek adına ve sadece search istekleri içinde olsa daha mantıklı olabilir. isteyen ekran alır kullanır gibi.